heimdall-middleware-spec/src/test/java/br/com/conductor/heimdall/middleware/util/helpermock/call/RequestMockTest.java

```diff
@@ -0,0 +1,50 @@
+package br.com.conductor.heimdall.middleware.util.helpermock.call;
+
+import org.junit.Test;
+import java.util.HashMap;
+
+import static org.junit.Assert.*;
+
+public class RequestMockTest {
+
+  @Test
+  public void testHeader() {
+    final RequestMock requestMock = new RequestMock();
+    assertEquals(new HashMap<String, String>(), requestMock.header().getAll());
+  }
+
+  @Test
+  public void testQuery() {
+    final RequestMock requestMock = new RequestMock();
+    assertEquals(new HashMap<String, String>(), requestMock.query().getAll());
+  }
+
+  @Test
+  public void testSetAppNameInput() {
+    final RequestMock requestMock = new RequestMock();
+    requestMock.setAppName("testName");
+    assertEquals("testName", requestMock.getAppName());
+  }
+
+  @Test
+  public void testSetBodyInput() {
+    final RequestMock requestMock = new RequestMock();
+    requestMock.setBody("testBody");
+    assertEquals("testBody", requestMock.getBody());
+  }
+
+  @Test
+  public void testSetUrl() {
+    final RequestMock requestMock = new RequestMock();
+    requestMock.setUrl("www.google.com");
+    assertEquals("www.google.com", requestMock.getUrl());
+  }
+
+  @Test
+  public void testSetPathParam() {
+    final RequestMock requestMock = new RequestMock();
+    assertEquals(null, requestMock.pathParam(""));
+    assertEquals(null, requestMock.pathParam("testPath"));
```

Comment:
Do these 3 tests cover all the lines in the method?
If so, are all 3 needed for that purpose?

Reply:
No the 3 tests do not cover any extra lines but they where added as you suggested adding more tests in a previous comment.
A single test would cover most of this function but the final part of the function is unreachable without reflection.

Reply:
OK I should have been more clear that by asking if more tests could be added, I meant tests which would add coverage as opposed to more tests to cover the same lines. If no more can be added without using reflection then I would keep with the minimum number.